src/compiler/utilities.ts

```diff
@@ -4062,6 +4062,12 @@ namespace ts {
         return node.kind === SyntaxKind.Identifier || isPropertyAccessEntityNameExpression(node);
     }
 
+    export function isDottedName(node: Expression): boolean {

```

Comment:
BTW, will this need to be updated for optional chains? Should a `t1?.assert(b)` be a thing?